hedera-node/docs/design/services/token-service/airdrops/airdrop-transaction.md

```diff
@@ -0,0 +1,208 @@
+ * - Each successful transfer that consumes an automatic association slot SHALL populate the
+ *   `automatic_association` field for the record.
+ * - Each pending transfer _created_ SHALL be added to the `pending_airdrops` field for the record.
+ * - Each pending transfer _updated_ SHALL be added to the `pending_airdrops` field for the record.
+ */
+ message TokenAirdropTransactionBody {
+/**
+     * A list of token transfers representing one or more airdrops.<br/>
+     * The sender for each transfer MUST have sufficient balance to complete the transfers.
+     * <p>
+     * All token transfers MUST successfully transfer tokens or create a pending airdrop
+     * for this transaction to succeed.<br/>
+     * This list MUST contain between 1 and 10 transfers, inclusive.
+     * <p>
+     * <blockquote>Note that each transfer of fungible/common tokens requires both a debit and
+     * a credit, so each _fungible_ token transfer MUST have _balanced_ entries in the
+     * TokenTransferList for that transfer.</blockquote>
+     */
+     repeated TokenTransferList token_transfers= 1;
+}
+```
+
+Add new RPC to `TokenService` :
+
+```protobuf
+   /**
+     * Airdrop one or more tokens to one or more accounts.<br/>
+     * This distributes tokens from the balance of one or more sending account(s) to the balance
+     * of one or more recipient accounts. Accounts will receive the tokens in one of four ways.
+     * <ul>
+     *   <li>An account already associated to the token to be distributed SHALL receive the
+     *       airdropped tokens immediately to the recipient account balance.</li>
+     *   <li>An account with available automatic association slots SHALL be automatically
+     *       associated to the token, and SHALL immediately receive the airdropped tokens to the
+     *       recipient account balance.</li>
+     *   <li>An account with "receiver signature required" set SHALL have a "Pending Airdrop"
+     *       created and MUST claim that airdrop with a `claimAirdrop` transaction.</li>
+     *   <li>An account with no available automatic association slots SHALL have a
+     *       "Pending Airdrop" created and MUST claim that airdrop with a `claimAirdrop`
+     *       transaction. </li>
+     * </ul>
+     * Any airdrop that completes immediately SHALL be irreversible. Any airdrop that results in a
+     * "Pending Airdrop" MAY be canceled via a `cancelAirdrop` transaction.<br/>
+     * All transfer fees (including custom fees and royalties), as well as the rent cost for the
+     * first auto-renewal period for any automatic-association slot occupied by the airdropped
+     * tokens, SHALL be charged to the account submitting this transaction.
+     */
+    rpc airdropTokens (Transaction) returns (TransactionResponse);
+```
+
+### Fee types
+
+The airdrop transaction would include several types of fees, a different set of them would be applicable, based on the conditions the airdrop is performed to. The list of fees is:
```

Comment:
Misuse of `would` here, and a slight improvement in wording.
```suggestion
The airdrop transaction will include several types of fees, a different set of them will be applicable, based on the characteristics of the airdrop. The list of fees is:
```

Reply:
Noted